src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java

```diff
@@ -0,0 +1,542 @@
+            fillValues(fieldStringMap, insertStatement, connection);
+            if (entityCount == 0) LOG.info(insertStatement.toString());
+            insertStatement.addBatch();
+            if (entityCount % INSERT_BATCH_SIZE == 0) {
+                LOG.info("Executing batch insert ({}/{}) for {}", entityCount, entityList.size(), childTableName);
+                int[] newIds = insertStatement.executeBatch();
+                LOG.info("Updated {}", newIds.length);
+            }
+            entityCount += 1;
+//            handleStatementExecution(connection, insertStatement, true);
+        }
+        // execute any remaining prepared statement calls
+        LOG.info("Executing batch insert ({}/{}) for {}", entityCount, entityList.size(), childTableName);
+        if (insertStatement != null) {
+            // If insert statement is null, an empty array was passed for the child table, so the child elements have
+            // been wiped.
+            int[] newIds = insertStatement.executeBatch();
+            LOG.info("Updated {}", newIds.length);
+        } else {
+            LOG.info("No inserts to execute. Empty array found in JSON for child table {}", childTableName);
+        }
+        LOG.info("Committing transaction.");
+        connection.commit();
+    }
+
+
+    @Override
+    public int delete(Integer id) throws SQLException {
+        Connection connection = dataSource.getConnection();
+        // Handle "cascading" delete or constraints on deleting entities that other entities depend on
+        // (e.g., keep a calendar from being deleted if trips reference it).
+        // FIXME: actually add "cascading"? Currently, it just deletes one level down.
+        deleteFromReferencingTables(tablePrefix, specTable, connection, id);
+        PreparedStatement statement = connection.prepareStatement(specTable.generateDeleteSql(tablePrefix));
+        statement.setInt(1, id);
+        LOG.info(statement.toString());
+        // Execute query
+        int result = statement.executeUpdate();
+        connection.commit();
+        if (result == 0) {
+            throw new SQLException("Could not delete entity");
+        }
+        // FIXME: change return message based on result value
+        return result;
+    }
+
+    @Override
+    public void commit() throws SQLException {
+        // FIXME: should this take a connection and commit it?
+    }
+
+    /**
+     * Delete entities from any referencing tables (if required). This method is defined for convenience and clarity, but
+     * essentially just runs updateReferencingTables with a null value for newKeyValue param.
+     */
+    private static void deleteFromReferencingTables(String namespace, Table table, Connection connection, int id) throws SQLException {
+        updateReferencingTables(namespace, table, connection, id, null);
+    }
+
+    /**
+     * Handles converting JSON object into a map from Fields (which have methods to set parameter type in SQL
```

Comment:
I had to read this several times to understand what it does. I think this is because there's some grammatical ambiguity that caused me to mentally parse it as "converting JSON object... to String values found in the JSON." Suggested replacement: "Convert a JSON object to a Map<Field, String> with one entry per attribute of the JSON object. This conversion is performed within the context of the supplied Table. We do this because we need to use methods on the Table's Fields to interpret and store the attribute values of the JSON object."